velox/type/Conversions.h

```diff
@@ -31,21 +31,31 @@ namespace facebook::velox::util {
 struct DefaultCastPolicy {
   static constexpr bool truncate = false;
   static constexpr bool legacyCast = false;
+  static constexpr bool sparkCast = false;
 };
 
 struct TruncateCastPolicy {
   static constexpr bool truncate = true;
   static constexpr bool legacyCast = false;
+  static constexpr bool sparkCast = false;
 };
 
 struct LegacyCastPolicy {
   static constexpr bool truncate = false;
   static constexpr bool legacyCast = true;
+  static constexpr bool sparkCast = false;
 };
 
 struct TruncateLegacyCastPolicy {
   static constexpr bool truncate = true;
   static constexpr bool legacyCast = true;
+  static constexpr bool sparkCast = false;
+};
+
+struct SparkCastPolicy {
```

Comment:
I don't see SparkCastPolicy used anywhere and I still see that Spark's cast test is being modified which suggests that this PR is breaking Spark's CAST. Let's not break it. I assume we need to extends exec::CastHooks API.

The new 'sparkCast' flag doesn't seem consistent with the existing flags (truncate and legacyCast) and is not documented. The existing flags are not documented either, which makes it hard to reason about them (CC: @gggrace14)

I believe there are 3 CAST behaviors: Presto, Spark, Legacy. One option is to have an enum with 3 values: kPresto, kSpark, kLegacy. Another option is to use per-feature flags, e.g. truncate, allow-yes-no-for-boolean, etc. I'm not sure which ones is better and whether they need to be template parameters, but it would be nice to come up with a consistent solution. I would think we don't need these as template parameters.

Reply:
I will start off consolidating the castHook Api then into the Presto/Spark/Legacy . Adding another template flag leads to too much branching in CastExpr : https://github.com/facebookincubator/velox/blob/b66f84cfbadc121290796b94741ad4e69eeee31c/velox/expression/CastExpr-inl.h#L705 

I will create another PR for above, and then subsequently work on this.